common/Measure-EM2-Current-Consumption-under-1.4uA-in-EFR32xG22.md

```diff
@@ -0,0 +1,269 @@
+Debug connectivity can be enabled by setting the EM2DBGEN field on the EMU_CTRL register, and will consume about 0.5 µA extra supply current. To reduce current consumption, comment out the line below.
+```c
+//Force PD0B to stay on EM2 entry. This allow debugger to remain connected in EM2
+//EMU->CTRL |= EMU_CTRL_EM2DBGEN; 
+```
+##### External flash
+The external flash “MX25R8035F” equipped in the BRD4182A radio board is standby by default and has a great amount of current, it should be shut off.
+```c
+ /* Disable external flash memory*/
+  MX25_init();
+  MX25_DP();
+```  
+MX25_init initializes the SPI Flash and calling MX25_DP sends the byte necessary to put the Flash into DP mode which should lower the current to about 0.5 µA.
+ 
+##### Voltage scaling
+Voltage scaling helps to optimize the energy efficiency of the system by operating at lower voltages when possible. Three supply voltage operating points are available:
+
+| VSCALE Setting | DECOUPLE Voltage | Operating Conditions |
+| :-------- | :-------- | :-------- |
+| VSCALE2 | 1.1 V | EM0/EM1 Operation up to 80 MHz, EM2 and EM3 |
+| VSCALE1 | 1.0 V | EM0/EM1 Operation up to 40 MHz, EM2 and EM3 |
+| VSCALE0 | 0.9 V | EM2 and EM3 Only |
+
+The voltage sale level for EM2 and EM3 is set using the EMU_CTRL_EMU23VSCALE field. The lowest sleep current will be obtained by setting EMU23VSCALE to VSCALE0.
+```c
+  EMU_EM23Init_TypeDef vsInit = EMU_EM23INIT_DEFAULT;
+  vsInit.vScaleEM23Voltage = emuVScaleEM23_LowPower;
+  EMU_EM23Init(&vsInit);
+```
+It has two Voltage scaling modes in EM2 and EM3, which are emuVScaleEM23_LowPower mode (vscale0) and emuVScaleEM23_FastWakeup (vscale2). Using emuVScaleEM23_LowPower will save approximately 1.3 µA comparing to emuVScaleEM23_FastWakeup.
+ 
+##### Radio RAM retention
+The EFR32xG22 device contains several blocks of SRAM for various purposes including general data memory (RAM) and various RF subsystem RAMS (SEQRAM, FRCRAM).
+Frame Rate Controller SRAM(FRCRAM) and all part of Sequencer SRAM(SEQRAM) may be powered down in EM2/EM3 if not required. To control retention of these areas, set FRCRAMRETNCTRL or SEQRAMRETNCTRL in SYSCFG_RADIORAMRETNCTRL to the desired value
+```c
+  /* Disable Radio RAM memories (FRC and SEQ) */
+  CMU_ClockEnable(cmuClock_SYSCFG, true);
+  SYSCFG->RADIORAMRETNCTRL = 0x103UL;
+```
+Disable different Radio RAM will result in different reduction. The results of current draw are listed in the experiments results in latter section.
+  
+##### DCDC
+The DC-DC buck converter covers a wide range of load currents, provides high efficiency in energy modes EM0, EM1, EM2 and EM3.
+```c
+  // Enable DC-DC converter
+  EMU_DCDCInit_TypeDef dcdcInit = EMU_DCDCINIT_DEFAULT;
+  EMU_DCDCInit(&dcdcInit);
+```
+ 
+##### GPIO
+All unconnected pins on the EFR32 should be configured to Disabled mode (high impedance, no pull resistor), and the reset state of the IO pins is disabled as well. This would be done by calling MX_25deinit().
+```c
```

Comment:
This should be done by set GPIO gpioModeDisabled, not by call MX_25deinit(), this function only set SPI relevant pin.